### DISCUSSION TITLE
Guidelines for the encoding of spatial data - CharacterString

### DISCUSSION THREAD

**Discussion by bfrichet3** [INSPIRE-MIF/helpdesk#99] [Q&A] [answered]
Dear Sir or Madam

I hope you are all right.

I have a question concerning the way of instatiating gmd:CharacterString objects in datasets' GML, for instance as I want to express the nationalCode attribute of one AdministrativeUnit object.

As I understand IR 1089/2010 I should instantiate one gmd:CharacterString tag inside a au:nationalCode. But I cannot find any GML files with such a structure and as I tested a GML from switzerland I didn't get any message about that problem. Should I understand that attributes with CharacterString ranges needn't any gco:CharacterString tag or is it something else (like a test that hasn't yet been implemented).
![image](https://user-images.githubusercontent.com/80756452/164461426-bcfe6190-7260-4196-83db-9f5c2ca26b87.png)

Regards
[testAUGMLSWITZERLAND.zip](https://github.com/INSPIRE-MIF/helpdesk/files/8531122/testAUGMLSWITZERLAND.zip)

![image](https://user-images.githubusercontent.com/80756452/164460179-a22fc64e-cd6c-4296-b994-d5168001677e.png)

**Comment by fabiovinci** [comment]
Dear @bfrichet3,

the encoding you used is right, the attributes with "CharacterString" dataType contain only text without additional child elements.

As yuo can see in the test report, the "Schema validation" test in the "Conformance class: INSPIRE GML application schemas, General requirements" is passed.

**Comment by bfrichet3** [comment]
Dear @fabiovinci

I am not totally convinced that is this way of writing is compliant with IR/ISO regulations. As you may know in the ISO 191xx norms, there is always an explicit gco:CharacterString tag, like in the ISO 19139 metadata records.

![image](https://user-images.githubusercontent.com/80756452/164480209-6fd4e539-6049-485c-b884-6c0813ebf702.png)


![image](https://user-images.githubusercontent.com/80756452/164480514-0e0fe7fd-2aad-4190-9136-b6833fe5a94f.png)


In my opinion, it is not clear if IR requires such an extra tag or not.

Regards

**Comment by fabiovinci** [comment, marked as answer]
Dear @bfrichet3,

the "_CharacterString_" dataType of the `nationalCode` attribute is derived from ISO19103, as you can see from the screenshot below:

![image](https://user-images.githubusercontent.com/22212406/164491814-3e8fcf28-0341-48f3-b369-eb9ca5544eee.png)

Following the profile of the conceptual model defined in the ISO 19100 series of International Standards implemented by GML (Annex D of OpenGIS® Geography Markup Language (GML) Encoding Standard), the “_CharacterString_” is implemented by xsd:string.

![image](https://user-images.githubusercontent.com/22212406/164497358-32f7d752-a6b4-4257-af0d-0b902b06df16.png)

Therefore in the XSD, this dataType is implemented as a simple type:

![image](https://user-images.githubusercontent.com/22212406/164497705-5d290675-ca12-46d7-8100-643ef43013d3.png)

Regarding the metadata, the [ISO19139](http://schemas.opengis.net/iso/19139/20070417/gco/basicTypes.xsd) defined an additional complex type

![image](https://user-images.githubusercontent.com/22212406/164498013-a944775c-c889-4917-ab62-8fe851a7d655.png)

which is used as dataType for [some metadata elements](http://schemas.opengis.net/iso/19139/20070417/gmd/citation.xsd):

![image](https://user-images.githubusercontent.com/22212406/164498853-e3e26547-d666-4cae-b134-f24a8e2560a0.png)

For these reasons, there is a different encoding in the GML and in the metadata for the same dataType.

**Comment by bfrichet3** [comment]
thank you for your clarification!
